### DISCUSSION TITLE
[FEATURE REQ] Custom Regex Parsing

### DISCUSSION THREAD

**Discussion by Gomeyy** [autobrr/autobrr#1671] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am currently experiencing an issue with Sonarr rejecting a release due to Unkown Series when I am sure that I have the title added in Sonarr. I am fairly certain this is happening because the name of the show is different on the release message when compared to the listing on Sonarr. 

**Describe the solution you'd like**
I would love to be able to modify the release message before it gets passed to the action using regex or something similar. This would allow me to implement a custom fix for the releases with issues.

**Describe alternatives you've considered**
Running a script which does the replace instead of pushing the release directly to Sonarr, which is what I am doing now. Would be nice to have the feature available in autobrr.

**Comment by KyleSanderson** [comment]
The background is it's some mangled Anime string from Animebytes, and the parsing library doesn't understand it (nor do I blame it, because the format is beyond not standard).

**Comment by varoOP** [comment]
Consider sending the torrent to torrent client first and have sonarr pick it up via label/category. The release's name will be mapped to sonarr's title via thexem. I use autobrr + sonarr for a lot of anime and this way is pretty seamless without having to use a script or any other form of custom parsing.

**Comment by gmcouto** [comment]
I know some trackers that do not use standard naming scheme also. It would be nice to parse and replace before handing it down to the action too.
I have been able to circumvent this by using a RSS pre-processor as a proxy. But would be nice for this capability be done in the radarr/sonarr action level, so it would work with IRC.
